homework/tests/calculatortest.spec.js

```diff
@@ -0,0 +1,338 @@
+// @ts-check
+const { test, expect } = require('@playwright/test');
+
+const data = [
+  'Prototype',
+  '1',
+  '2',
+  '3',
+  '4',
+  '5',
+  '6',
+  '7',
+  '8',
+  '9'
+]
+
+
+data.forEach(version => {
+  test.describe(version + ': Add', () => {
+    test('Adding 2 and 3 results in 5', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('2');
+      await page.locator('#number2Field').type('3');
+      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('5');
+    });
+  });
+});
+
```

Comment:
It is good that you thought of adding integers and non-integers, but in the homework description it was described "**Tests should cover main functions, critical flows.**" , so that being said, only single test for adding numbers (integers or non-integers) would be enough without going into edge-cases. Please try to think about your automation strategy and remove tests which are repetitive (like adding integers and non-integers). 

However, let's leave 1 repetitive scenario. For this case, separate function should be created and called with different parameters. So instead of deleting all repetitive scenarios, please leave  2 tests regarding Add function and when you call those tests use function. I am attaching a not finished code snippet which I hope will help you to figure this out.

```
    async checkAddFunction(pageUrl, buildVersion, numberOne, numberTwo) {
        await page.goto(pageUrl);
        await page.selectOption('#selectBuild', {label: buildVersion});
        await page.locator('#number1Field').type(numberOne);
        ...
        ...
    }
```

```
 test('Adding -3 and 0 results in -3', async ({page}) => {
await checkAddFunction("https://testsheepnz.github.io/BasicCalculator", version, "3" ...);


            await expect(page.locator('#numberAnswerField')).toHaveValue('-3');
        });
    });
```

Disclaimer: I did not test this code and it might need some improvement, it is just for you to get an idea how it should be done.